### DISCUSSION TITLE
bug? enable_surface_pick

### DISCUSSION THREAD

**Discussion by lovecoding2023** [pyvista/pyvista#3703] [Q&A] [answered]
Hello,
I tried to switch enable_surface_picking multiple times, and each time I turned it on, the callback would accumulate once. How to delete the previous callback time? is it bug? thanks
```
plotter.enable_surface_picking(callback=mycallback, tolerance=0.01,pickable_window=False, show_message=False,left_clicking=True,show_point=False)
......
plotter.disable_picking()
......
plotter.enable_surface_picking(callback=mycallback, tolerance=0.01,pickable_window=False, show_message=False,left_clicking=True,show_point=False)

......

def mycallback(point):
print('point',point)
```
result:
```
point [ 0.39812151  1.63781739 12.40211855]
point [ 0.39812151  1.63781739 12.40211855]
```
*If the callback times in **enable_point_picking** are turned on before, they will also be accumulated in the new callback of **enable_surface_picking**.

**Comment by banesullivan** [comment, marked as answer]
Ah hm. It seems like a bug where the event observers are not getting cleared when picking is disabled. There is not likely a quick fix for this. I've added this to https://github.com/pyvista/pyvista/issues/3100 where hopefully we can tackle this while improving the picking in other ways.

Thanks for raising!

**Reply by banesullivan** [reply]
This should be solved by #4254
